src/transactions/test/MergeTests.cpp

```diff
@@ -726,6 +733,12 @@ TEST_CASE("merge", "[tx][merge]")
                     ltx.commit();
                 }
 
+                SECTION("also with sponsored signer")
```

Comment:
It looks like the sponsored account with no sponsored signer test isn't run anymore. You can verify this by adding the code below after the section you added, which shows that the signer is always sponsored. Was this your intention? This caught my eye because the `SECTION` usage is different from what we normally do. 

```
{
       LedgerTxn ltx(app->getLedgerTxnRoot());
       auto signer = makeSigner(getAccount("S1"), 1);
       checkSponsorship(ltx, acc1, signer.key, 2, &sponsoringAcc.getPublicKey());
}
```

Reply:
@sisuresh It looks to me like this is working (eg. if you change line 732 to read `nullptr` instead of `&sponsoringAcc.getPublicKey()`, it fails). Am I misunderstanding you?

Reply:
Yeah the test passes. My concern is that the test always merges a sponsored account **with** a sponsored signer. Previously, the test merged a sponsored account **without** a sponsored signer.

I'm not sure if this was done intentionally, or if an assumption was made about `SECTION("with sponsored account")` running twice and `SECTION("also with sponsored signer")` only running once. Does that make sense?

Reply:
Ah yes I understand what @sisuresh is saying, and his comment is right. @MonsieurNicolas your change has effectively removed the test that merges a sponsored account without a sponsored signer, because the "also with sponsored signer" always executes. Can you restore it so that it actually does execute both variations?

Reply:
Oh, I thought that when writing sections like this it would explore all possible branches. I guess not.
Anyways, as @sisuresh already wrote this test, we should just use his version instead of mine

Reply:
@MonsieurNicolas Your change was cleaner than mine so I just fixed the section issue from your commit. Here's the PR  https://github.com/stellar/stellar-core/pull/2682.